### PR TITLE
Allow users to generate a unique token to create their own shopping list

### LINKED ISSUE #3: 3. As a user, I want to set up a new shopping list so I can start tracking purchased items
## Summary 
A shopping list is a set of items associated with a specific three-word token. We need to allow users to start new lists, so they can save the things they need to buy. To start a new list, we will need to do a few things:

1. Generate a new, unique token
2. Save the token to `localStorage`
3. Show the user the list view

Up to this point, the app has rendered the items in the list associated with the token `my test list`. Before we can allow users to make *new* lists, we have to stop showing them this test list.

## Acceptance criteria
- [x] `@the-collab-lab/shopping-list-utils` is added as a dependency to the project.
- [x] The string `my test list` in `App.jsx` is replaced with `null`, so users are not automatically subscribed to any list.

If a user **doesn’t** already have a token:
- [x] A button in the `Home` component allows them to create a new list
- [x] Clicking the button generates a new token and saves it to `localStorage` using the `setListToken` function in `App.jsx`.
- [x] Once the token has been created and saved, the user is redirected to the `List` view.

If a user **does** already have a token:
- [x] They are automatically redirected to the `List` view.

## Notes
- The  `generateToken` function from the `shopping-list-utils` package should be used to generate the token. More information on that package [its documentation](https://www.npmjs.com/package/@the-collab-lab/shopping-list-utils). Importing it will look something like this:
``` js
import { generateToken } from '@the-collab-lab/shopping-list-utils';
```
- The `setListToken` function takes **one** argument: the new list token.
- Eventually the welcome/home page will look like [this wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/welcome-screen.png). It’s good for your whole team to look at the wireframe now, but be sure not to try to implement all of the visible functionality at once. There will be future stories to add the other features such as joining an existing list.
- For this story, it's sufficient just to have a simple "create list" button. Although you are not required to save the token to Firestore, some teams have opted to also save the token to Firestore when generating a token.

### PR DESCRIPTION
**Pull request by KarenJParker77**
## Description
This pull request relates to the issue of allowing users to create a new shopping list so they can begin tracking their purchases. 

## Create List button
We added a button in the Home view, which, when clicked generates a three word token and stores it in local storage. The user is then taken to the List view. 

## Redirect users who have a token already
If they access the site already having a token in local storage, they are taken directly to the List view. 

## Things we liked :)
We used the Link and Navigate components from React Router Dom, which made for clean, straightforward code.

## Related Issue
Closes https://github.com/the-collab-lab/tcl-59-smart-shopping-list/issues/3

## Acceptance Criteria

 

- [x] @the-collab-lab/shopping-list-utils is added as a dependency to the project.
- [x]  The string my test list in App.jsx is replaced with null, so users are not automatically subscribed to any list.

If a user doesn’t already have a token:

- [x] A button in the Home component allows them to create a new list
- [x]  Clicking the button generates a new token and saves it to localStorage using the setListToken function in App.jsx.
- [x]  Once the token has been created and saved, the user is redirected to the List view.

If a user does already have a token:

- [x]  They are automatically redirected to the List view.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|    ✓ | :link: Update dependencies |
|     | :scroll: Docs              |

## Testing Steps / QA Criteria

Please pull this branch down with the command- git pull origin sg-kp-new-list and check that branch out with git checkout sg-kp-new-list.
Then npm start to launch the app. Please note that @the-collab-lab/shopping-list-utils has been added as a dependency to the project so you will need to install that locally.
Please click on the 'create list' button and ensure it re-directs you to List view, checking also that there's a three word token in localStorage (Inspect => application => local storage). Ensure that because you now have a token, you are redirected to the List view when you open the app.

### CODE REVIEW

**Review by cassie202 — COMMENTED** [review]
Great Job! Well done..
However, creating a token works perfectly but if I delete a token from the local storage and refresh the page, it still leaves me in the list page.

I saw in the Acceptance criteria that without a token, I should be taken to the home page to create a token.

**Comment by mikeyxx** [comment]
Hi Karen and Sandra, great job guys🥳

Just to add to Cassandra's comment, when you delete a token and refresh, the user should be redirected to the home page to create a new token. To get that working, you just need to replicate what you did in the home route for the list route too😉

**Comment by mikeyxx** [comment]
> Hi Karen and Sandra, great job guys🥳
> 
> Just to add to Cassandra's comment, when you delete a token and refresh, the user should be redirected to the home page to create a new token. To get that working, you just need to replicate what you did in the home route for the list route too😉

Same thing for the AddItem page. Basically a user shouldn't be able to access any of the other pages if they don't have a listToken

File: package-lock.json
Comment (mikeyxx) [inline package-lock.json:1]: Looks good and works fine. All criteria have been met👍🏽

**Review by mikeyxx — APPROVED** [review]

**Comment by mikeyxx** [comment]
Looks good and works fine. All criteria have been met👍🏽

Well done ladies😁

**Review by cassie202 — APPROVED** [review]
It works perfectly now!

**Review by amanolar94 — APPROVED** [review]
Nice effort!

**Pull request merged**